packages/main/src/components/emoji/Emoji.jsx

```diff
@@ -0,0 +1,535 @@
+        d="M8.4,0a8.4,8.4,0,0,0,0,16.8h0a8.4,8.4,0,0,0,8.4-8.4"
+        transform="translate(2.4 4.8)"
+        fill="none"
+        stroke="#000"
+      />
+      <path
+        id="Oval-2"
+        data-name="Oval"
+        d="M0,0A4.807,4.807,0,0,0,1.042,1.557,4.785,4.785,0,0,0,4.436,2.963h0A4.785,4.785,0,0,0,7.83,1.557,4.807,4.807,0,0,0,8.872,0"
+        transform="translate(6.364 15.037)"
+        fill="none"
+        stroke="#000"
+      />
+      <circle
+        id="Oval-3"
+        data-name="Oval"
+        cx="1.2"
+        cy="1.2"
+        r="1.2"
+        transform="translate(7.2 9.6)"
+        fill="#fff"
+        stroke="#000"
+      />
+      <circle
+        id="Oval-4"
+        data-name="Oval"
+        cx="1.2"
+        cy="1.2"
+        r="1.2"
+        transform="translate(12 9.6)"
+        fill="#fff"
+        stroke="#000"
+      />
+      <path
+        id="Line"
+        d="M1.2,0V7.2"
+        transform="translate(18 1.2)"
+        fill="none"
+        stroke="#000"
+      />
+      <path
+        id="Line-2"
+        data-name="Line"
+        d="M0,1.2H7.2"
+        transform="translate(15.6 3.6)"
+        fill="none"
+        stroke="#000"
+      />
+    </svg>
+  );
+
+  const commentIcon = (
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+      <path d="M17,9H7a1,1,0,0,0,0,2H17a1,1,0,0,0,0-2Zm-4,4H7a1,1,0,0,0,0,2h6a1,1,0,0,0,0-2ZM12,2A10,10,0,0,0,2,12a9.89,9.89,0,0,0,2.26,6.33l-2,2a1,1,0,0,0-.21,1.09A1,1,0,0,0,3,22h9A10,10,0,0,0,12,2Zm0,18H5.41l.93-.93a1,1,0,0,0,0-1.41A8,8,0,1,1,12,20Z" />
+    </svg>
+  );
+
+  useEffect(() => {
+    if (display) document.body.style.overflow = "hidden";
+    if (!display) document.body.style.overflow = "scroll";
+  }, [display]);
+
+  useEffect(() => {
+    if (!isOpen) {
+      setNumber(null);
+    }
+  }, [isOpen]);
+
+  const setEmojiName = e => {
+    setState({ name: e.target.value });
+  };
+
+  const handleImageChange = e => {
+    if (imageRef.current.files[0]) {
+      let fileReader = new FileReader();
+
+      fileReader.onload = function (e) {
+        avatarRef.current.src = e.target.result;
+        _avatarRef.current.src = e.target.result;
+        setEmojiImage(e.target.result);
+      };
+
+      fileReader.readAsDataURL(imageRef.current.files[0]);
+      const imageReader = e.target.files[0];
+
+      setFileName(imageRef.current.files[0].name);
+
+      setFormData(formData.append("image", imageReader));
+    }
+  };
+
+  useEffect(() => {
+    //check for returned data from upload
+  }, [state.name]);
+
+  const handleSave = e => {
+    setDisplay(false);
+
+    //get the user data from session storage
+    const getUser = () => {
+      const result = JSON.parse(sessionStorage.getItem("user") || null);
+      return result;
+    };
+    //get the currentWorkspace from local storage
+    const getCurrentWorkspace = () => {
+      const currentWorkspace = localStorage.getItem("currentWorkspace") || null;
+      return currentWorkspace;
```

Comment:
you defined these functions somewhere else so there is no reason to rewrite it here